torchvision/prototype/transforms/_type_conversion.py

```diff
@@ -33,3 +37,28 @@ def extra_repr(self) -> str:
             return ""
 
         return f"num_categories={self.num_categories}"
+
+
+class ToImageTensor(Transform):
+    def __init__(self, *, copy: bool = False) -> None:
+        super().__init__()
+        self.copy = copy
+
+    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
+        if isinstance(input, (features.Image, PIL.Image.Image, np.ndarray)) or is_simple_tensor(input):
+            output = F.to_image_tensor(input, copy=self.copy)
+            return features.Image(output)
```

Comment:
When I first read `to_image_tensor()` I expected you would be returning `features.Image` not `Tensor` but then remembered that all other `*_image_tensor()` methods return Tensors for JIT-scriptability...

But then isn't it weird that `ToImageTensor` return a `features.Image`? I'm not actually asking you to change it, just trying to figure out how we want this to work. 

Could you elaborate when people would use this transform? The way I understand it is that the datasets will return I `feature.Image` entries, so the use of this transform will be limited to only when someone has read a PIL image manually and want to convert them to Image. Right?

Reply:
> But then isn't it weird that `ToImageTensor` return a `features.Image`? I'm not actually asking you to change it, just trying to figure out how we want this to work.

Maybe a little unexpected, but I wouldn't say weird. Given that `features.Image` is a superset of what `torch.Tensor` provides, users should not feel a difference. No strong opinion though since all other transforms accept both types. Maybe it is a nice way to "promote" the new API.

> Could you elaborate when people would use this transform? The way I understand it is that the datasets will return I `feature.Image` entries, so the use of this transform will be limited to only when someone has read a PIL image manually and want to convert them to Image. Right?

Correct. Neither `ToImageTensor` nor `ToImagePIL` will be needed if users stay in "`torchvision` land". Depending on their own code, they might use these though.

Reply:
Sounds good. I'm OK adopting this approach for now. If you fix the conflicts, breakages and remaining TODOs we should be able to merge.